server/odc-service/src/main/java/com/oceanbase/odc/service/flow/FlowTaskInstanceService.java

```diff
@@ -380,27 +381,48 @@ public List<BinaryDataResult> download(@NonNull Long flowInstanceId, String targ
                 taskEntityOptional::isPresent);
         TaskEntity taskEntity = taskEntityOptional.get();
         if (!dispatchChecker.isTaskEntityOnThisMachine(taskEntity)) {
-            /**
-             * 任务不在该节点上，需要进行请求转发
-             */
-            ExecutorInfo executorInfo = JsonUtils.fromJson(taskEntity.getExecutor(), ExecutorInfo.class);
-            DispatchResponse dispatchResponse =
-                    requestDispatcher.forward(executorInfo.getHost(), executorInfo.getPort());
-            HttpHeaders headers = dispatchResponse.getResponseHeaders();
-            if (headers == null) {
-                return Collections
-                        .singletonList(new ByteArrayDataResult("download.data", dispatchResponse.getContent()));
-            }
-            String fileName = headers.getContentDisposition().getFilename();
-            if (fileName == null) {
-                return Collections
-                        .singletonList(new ByteArrayDataResult("download.data", dispatchResponse.getContent()));
-            }
-            return Collections.singletonList(new ByteArrayDataResult(fileName, dispatchResponse.getContent()));
+            // The task is not executing on current machine, need to forward the request
+            return dispatchRequest(taskEntity);
         }
         return internalDownload(taskEntity, targetFileName);
     }
 
+    public List<BinaryDataResult> downRollbackPlanResult(@NonNull Long flowInstanceId) throws IOException {
```

Comment:
Is there any authentication logic required here?

Reply:
Same as other file download API, there is no permission check yet. The URL is added in `OrganizationAuthenticationInterceptor` as a whitelist.